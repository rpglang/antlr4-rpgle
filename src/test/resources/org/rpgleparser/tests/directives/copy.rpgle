     C/COPY MBR1
     I/INCLUDE SRCFIL,MBR2
     O/COPY SRCLIB/SRCFIL,MBR3
     O/INCLUDE "SRCLIB!"/"SRC>3","MBRÂ¬3"
     O/COPY /dir1/dir2/file.rpg
     O/COPY /dir1/dir2/file
     O/COPY dir1/dir2/file.rpg
     O/COPY "ifs file containing blanks"
     O/COPY 'ifs file containing blanks'