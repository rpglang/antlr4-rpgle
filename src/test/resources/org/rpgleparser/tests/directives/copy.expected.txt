/*===TOKENS===*/
DIRECTIVE                <C/>
DIR_COPY                 <COPY>
DIR_OtherText            <MBR1>
EOL                      <>
DIRECTIVE                <I/>
DIR_INCLUDE              <INCLUDE>
DIR_OtherText            <SRCFIL>
DIR_OtherText            <MBR2>
EOL                      <>
DIRECTIVE                <O/>
DIR_COPY                 <COPY>
DIR_OtherText            <SRCLIB>
DIR_Slash                </>
DIR_OtherText            <SRCFIL>
DIR_OtherText            <MBR3>
EOL                      <>
DIRECTIVE                <O/>
DIR_INCLUDE              <INCLUDE>
StringLiteralStart       <">
StringContent            <SRCLIB!>
StringLiteralEnd         <">
DIR_Slash                </>
StringLiteralStart       <">
StringContent            <SRC>3>
StringLiteralEnd         <">
StringLiteralStart       <">
StringContent            <MBR¬3>
StringLiteralEnd         <">
EOL                      <>
DIRECTIVE                <O/>
DIR_COPY                 <COPY>
DIR_Slash                </>
DIR_OtherText            <dir1>
DIR_Slash                </>
DIR_OtherText            <dir2>
DIR_Slash                </>
DIR_OtherText            <file.rpg>
EOL                      <>
DIRECTIVE                <O/>
DIR_COPY                 <COPY>
DIR_Slash                </>
DIR_OtherText            <dir1>
DIR_Slash                </>
DIR_OtherText            <dir2>
DIR_Slash                </>
DIR_OtherText            <file>
EOL                      <>
DIRECTIVE                <O/>
DIR_COPY                 <COPY>
DIR_OtherText            <dir1>
DIR_Slash                </>
DIR_OtherText            <dir2>
DIR_Slash                </>
DIR_OtherText            <file.rpg>
EOL                      <>
DIRECTIVE                <O/>
DIR_COPY                 <COPY>
StringLiteralStart       <">
StringContent            <ifs file containing blanks>
StringLiteralEnd         <">
EOL                      <>
DIRECTIVE                <O/>
DIR_COPY                 <COPY>
StringLiteralStart       <'>
StringContent            <ifs file containing blanks>
StringLiteralEnd         <'>
/*===TREE===*/
(r
  (statement (directive C/ (dir_copy COPY (copyText MBR1)) ))
  (statement
    (directive I/ (dir_include INCLUDE (copyText SRCFIL) (copyText MBR2)) )
  )
  (statement
    (directive
      O/
      (dir_copy COPY (copyText SRCLIB) / (copyText SRCFIL) (copyText MBR3))
      
    )
  )
  (statement
    (directive
      O/
      (dir_include
        INCLUDE
        (copyText " SRCLIB! ")
        /
        (copyText " SRC>3 ")
        (copyText " MBR¬3 ")
      )
      
    )
  )
  (statement
    (directive
      O/
      (dir_copy COPY / (copyText dir1) / (copyText dir2) / (copyText file.rpg))
      
    )
  )
  (statement
    (directive
      O/
      (dir_copy COPY / (copyText dir1) / (copyText dir2) / (copyText file))
      
    )
  )
  (statement
    (directive
      O/
      (dir_copy COPY (copyText dir1) / (copyText dir2) / (copyText file.rpg))
      
    )
  )
  (statement
    (directive O/ (dir_copy COPY (copyText " ifs file containing blanks ")) )
  )
  (statement
    (directive O/ (dir_copy COPY (copyText ' ifs file containing blanks ')) <EOF>)
  )
)
/*======*/